packages/advisor-components/src/RuleRating/RuleRating.tsx

```diff
@@ -0,0 +1,52 @@
+import './RuleRating.scss';
+
+import React, { useState } from 'react';
+import { useIntl } from 'react-intl';
+
+import { Button } from '@patternfly/react-core/dist/js/components/Button/Button';
+import OutlinedThumbsDownIcon from '@patternfly/react-icons/dist/js/icons/outlined-thumbs-down-icon';
+import OutlinedThumbsUpIcon from '@patternfly/react-icons/dist/js/icons/outlined-thumbs-up-icon';
+import ThumbsDownIcon from '@patternfly/react-icons/dist/js/icons/thumbs-down-icon';
+import ThumbsUpIcon from '@patternfly/react-icons/dist/js/icons/thumbs-up-icon';
+import messages from '../messages';
+import { Rating } from '../types';
+
+interface RuleRatingProps {
+  ruleId: string;
+  ruleRating: Rating;
+  onVoteClick: (ruleId: string, calculatedRating: Rating) => unknown;
+}
+
+const RuleRating: React.FC<RuleRatingProps> = ({ ruleId, ruleRating, onVoteClick }) => {
+  const intl = useIntl();
+  const [rating, setRating] = useState(ruleRating);
+  const [submitted, setSubmitted] = useState(false);
+  const [thankYou, setThankYou] = useState(intl.formatMessage(messages.feedbackThankYou));
+
+  const updateRuleRating = async (newRating: Rating) => {
+    const calculatedRating = rating === newRating ? 0 : newRating;
+    try {
+      await onVoteClick(ruleId, calculatedRating);
+      setRating(calculatedRating);
+      setSubmitted(true);
+      setTimeout(() => setThankYou(''), 3000);
```

Comment:
Wouldn't it be better to use `debounce` here? So we won't send multiple requests?

Reply:
@karelhala, isn't only one network request made by calling `onVoteClick`? Why do you say there are multiple?

Reply:
What if the user starts to click multiple times on the button? I would have guessed we'd show the thank you message every 3 seconds. With debounce we'd show the `thank you` after the final button press.

Reply:
That is a nice point, thank you. Reworked in https://github.com/RedHatInsights/frontend-components/pull/1493/commits/669206b9399b2bab1651533197aa94fba6ffbf13